[Winlogbeat] Add permissions required for local user account (#37176)
elastic/beats: winlogbeat/docs/getting-started.asciidoc

Reviewer: To match the existing style, please wrap the lines at ~ 80 chars (or whatever this file uses).

Author: updated as suggested

+{beatname_lc}.yml+ and specify a user who is

> <<privileges-to-publish-events,authorized to publish events>>.

Author: I updated them, but in the PR, I don't see them getting transformed into "Winlogbeat" and "winlogbeat". Is that applied during the CI?

> 

Reviewer: ```suggestion
=== Using a local non-Administrator account to run Winlogbeat

```

It think we should give this section a title and move it out of step 5. I'm open to new location suggestions, but I was thinking to move it to the bottom of this file. And then in step 1 (where we install the service) we link to this section.

Author: Added at the end and a link in step1.

> By default, the `winlogbeat` service runs as `Local System` account. If you want to run `winlogbeat` service as a local user account which is not an Administrator, follow the below steps. By following these steps, your local user account will be granted `Log on as a service` security policy and made part of group `Event Log Readers` to read the event log. (You can skip these steps if you want to continue running Winlogbeat as an Administrator.)

Reviewer: ```suggestion
By default, the `{beatname_uc}` service runs as the `LocalSystem` account. If you want to run the `{beatname_uc}` service as a local user account that is not an Administrator, then follow the below steps. The local user account must be granted `Log on as a service` in the security policy and be made part of the `Builtin\Event Log Readers` group to read the event log.
```

I changed to account names to match the names in https://learn.microsoft.com/en-us/windows-server/identity/ad-ds/manage/understand-security-identifiers.

Author: Updated as suggested

> By default, the `winlogbeat` service runs as `Local System` account. If you want to run `winlogbeat` service as a local user account which is not an Administrator, follow the below steps. By following these steps, your local user account will be granted `Log on as a service` security policy and made part of group `Event Log Readers` to read the event log. (You can skip these steps if you want to continue running Winlogbeat as an Administrator.)

Author: The service name is `winlogbeat` in lowercase when installed.

<img width="759" alt="Screenshot 2023-11-23 at 2 16 10 PM" src="https://github.com/elastic/beats/assets/11301409/49b768b8-04ce-489d-a6b7-bff7253af557">

> By default, the `winlogbeat` service runs as `Local System` account. If you want to run `winlogbeat` service as a local user account which is not an Administrator, follow the below steps. By following these steps, your local user account will be granted `Log on as a service` security policy and made part of group `Event Log Readers` to read the event log. (You can skip these steps if you want to continue running Winlogbeat as an Administrator.)

Author: updated as suggested below

> . Open the Services Management console in Windows, if not already open by running this command:

Reviewer: ```suggestion
. Open the Services Management console with this command:
```

"on Windows" is implicit for Winlogbeat

"if not already open" - that seems unnecessary.

Author: updated as suggested

[source,shell]
----------------------------------------------------------------------
PS C:\Program Files\Winlogbeat> services.msc
----------------------------------------------------------------------
. Right-click on service named `winlogbeat` and select `Properties`
. Under `Log On` tab, select `This account:` and browse for your local account user that you want to run `winlogbeat` service.
. Enter local user account's password and click `Apply`.
. Search and open `Local Group Policy Editor` in Windows search or run `gpedit.msc` from Powershell.
. Navigate to path: `Computer Settings -> Security Settings -> Local Policies` and open `User Rights Assignment` under it.
. Inside `User Rights Assignment`, add your local user account to the policy named `Log on as a service`. This should allow your local user account log on as a service.
. Open `Local Users and Group Manager` by running `lusrmgr.msc` in Powershell.
. Under `Users`, right-click on your local account user and open `Properties`.
. Select `Member of` tab and click on `Add..`
. Find and select the group named `Event Log Readers` and click `Apply`. This should allow your local account user to read the event log.

> NOTE: A force apply maybe required to apply the policy selected from `Local Group Policy Editor`. Run `gpupdate /force` from Powershell if you are having trouble starting up the `winlogbeat` service in next step.

Reviewer: For local policy changes this is not needed.

> If you modify the security settings on your local device by using the local security policy, then you're directly modifying the settings on your device. Therefore, the settings take effect immediately, ...

https://learn.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/administer-security-policy-settings

Author: Removed the note

include::{libbeat-dir}/shared/opendashboards.asciidoc[tag=open-dashboards]

[float]
[[local-user-account-setup]]
=== Using a local non-Administrator account to run Winlogbeat

By default, the +{beatname_uc}+ service runs as the `Local System` account. 
If you want to run the +{beatname_uc}+ service as a local user account 

> that is not an Administrator, then follow the below steps. The local user account 

Reviewer 2: ```suggestion
that is not an Administrator, then follow the steps below. The local user account 
```

must be granted `Log on as a service` in the security policy 
and be made part of the `Builtin\Event Log Readers` group to read the event log. 

. Open the Services Management console with this command:
[source,shell]
----------------------------------------------------------------------
PS C:\Program Files\Winlogbeat> services.msc
----------------------------------------------------------------------

> . Right-click on service named +{beatname_lc}+ and select `Properties`

Reviewer 2: This appears to break the list numbering; "Open the serv..." is at 1. and then "Right-click on ser..." is at 1. again. Asciidoc presumably has something like four space indents to make text belong to the preceding list item.

> . Under `Log On` tab, select `This account:` and browse for your local account user 
> that you want to run {beatname_uc} service.

Reviewer 2: ```suggestion
. Under `Log On` tab, select `This account:` and browse for the local account user 
that you want to run {beatname_uc} service as.
```
or `s/as/under/`.

. Enter local user account's password and click `Apply`.
. Search and open `Local Group Policy Editor` in Windows search or 
run `gpedit.msc` from Powershell.

> . Navigate to path: `Computer Settings -> Security Settings -> Local Policies` 

Reviewer 2: flourish?
```suggestion
. Navigate to path: `Computer Settings → Security Settings → Local Policies` 
```

<img width="852" alt="Screenshot 2023-11-28 at 16 51 33" src="https://github.com/elastic/beats/assets/90160302/c717afdf-28ed-4c9d-85ee-4a371e1dfe1e">

and open `User Rights Assignment` under it.
. Inside `User Rights Assignment`, add your local user account to the policy named 
`Log on as a service`. This should allow your local user account log on as a service.
. Open `Local Users and Group Manager` by running `lusrmgr.msc` in Powershell.
. Under `Users`, right-click on your local account user and open `Properties`.

> . Select `Member of` tab and click on `Add..`

Reviewer 2: Does the menu have only two dots, or is it an ellipsis ('…')?
